File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <bits/stdc++.h>
using namespace std;

string target;
int cnt, n;

void dfs(string s) {
    cnt++;
    
    if(s == target) {
        n = cnt;
        return;
    }
    if(s.size() >= 5) return;
    
    for(char c : {'A', 'E', 'I', 'O', 'U'})
        dfs(s + c);
}

int solution(string word) {
    target = word;
    dfs("");
    return n-1;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
char_list = ['A', 'E', 'I', 'O', 'U']

answer = 0
cnt = -1

def DFS(cur, target):
    global answer, cnt

    if len(cur) > 5:
        return
    
    cnt += 1

    if (cur == target):
        answer = cnt
        return
    
    for c in char_list:
        DFS(cur + c, target)
    
def solution(word):
    DFS("", word)
    return answer